agenda/tests/Feature/ContactTest.php

```diff
@@ -0,0 +1,836 @@
+            ->assertHeader('content-type', 'application/json')
+            ->assertStatus(200)
+            ->assertJson([
+                'success' => true,
+                'request' => route('contacts.show', $contact->id),
+                'method'  => 'GET',
+                'code'    => 200,
+                'data'    => [
+                    'id'      => $this->user->contacts->first()->id,
+                    'user_id' => $this->user->id,
+                    'name'    => $this->user->contacts->first()->name,
+                    'phones'  => [
+                        [
+                            'id'           => $phone->id,
+                            'phone_number' => $phone->phone_number
+                        ],
+                    ],
+                    'adresses' => [
+                        [
+                            'id'           => $address->id,
+                            'street_name'  => $address->street_name,
+                            'number'       => $address->number,
+                            'complement'   => $address->complement,
+                            'neighborhood' => $address->neighborhood,
+                            'city'         => $address->city,
+                            'state'        => $address->state,
+                            'postal_code'  => $address->postal_code,
+                            'country'      => $address->country
+                        ],
+                    ],
+                    'tags' => null
+                ],
+            ], true);
+    }
+
+    /**
+     * Retorna erro ao tentar realizar busca dos detalhes do contato que possui o
+     * id informado, mas esse contato não existe
+     */
+    public function testReturnErrorWhenShowDetailsOfContactThatDoesntExists()
+    {
+        $contact = factory(Contact::class)->create([
+            'user_id' => $this->user->id
+        ]);
+
+        $contact->delete();
+
+        $this->get(route('contacts.show', $contact->id))
+            ->assertHeader('content-type', 'application/json')
+            ->assertStatus(200)
+            ->assertJson([
+                'success' => false,
+                'request' => route('contacts.show', $contact->id),
+                'method'  => 'GET',
+                'code'    => 200,
+                'data'    =>  null,
+            ], true)
+            ->assertJsonStructure(['errors'])
+            ->assertJsonFragment([
+                'code' => 14
+            ]);
+
+        $this->assertEmpty($this->user->contacts);
+    }
+
+    /**
+     * Retorna erro ao tentar realizar busca dos detalhes do contato que possui o
+     * id informado, mas usuário não tem autenticação
+     */
+    public function testReturnErrorWhenShowDetailsOfContactWhenUserIsUnauthorized()
+    {
+        $this->withHeaders(['Authorization' => $this->generateUnauthorizedToken()]);
+
+        $contact = factory(Contact::class)->create([
```

Comment:
Não está errado, mas nesse teste nem precisa criar o contato em si. Criar um id fake para usar na request ja basta...
Exemplo : $idContact = $this->faker->uuid;